### PR TITLE
feat: support splitting certs and private key

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Supports writing cert and key in separate files in addition to the default behavior of single file (maintains backward compatibility)

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/933

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
Ready for initial review, i'll add docs in a separate commit!

File: cmd/main.go [hunk 1]
```diff
@@ -46,6 +45,10 @@ var (
 
 	metricsBackend = flag.String("metrics-backend", "Prometheus", "Backend used for metrics")
 	prometheusPort = flag.Int("prometheus-port", 8898, "Prometheus port for metrics backend")
+
+	constructPEMChain              = flag.Bool("construct-pem-chain", true, "explicitly reconstruct the pem chain in the order: SERVER, INTERMEDIATE, ROOT")
+	writeCertAndKeyInSeparateFiles = flag.Bool("write-cert-and-key-in-separate-files", false,
```
Comment (aramase) [inline cmd/main.go:50]: I'm not a big fan of the long flag name but it's descriptive. I'm open to suggestions on this one.

File: .pipelines/templates/e2e-test.yaml [hunk 2]
```diff
@@ -72,3 +72,5 @@ steps:
         CI_KIND_CLUSTER: ${{ parameters.ciKindCluster }}
       ${{ if parameters.isArcTest }}:
         IS_ARC_TEST: ${{ parameters.isArcTest }}
+      ${{ if parameters.testReleasedVersion }}:
+        GINKGO_SKIP: WriteCertAndKeyInSeparateFiles
```
Comment (nilekhc) [inline .pipelines/templates/e2e-test.yaml:76]: Why are we skipping this?

File: pkg/provider/provider.go [hunk 3]
```diff
@@ -858,3 +878,29 @@ func fetchCertChains(data []byte) ([]byte, error) {
 	}
 	return pemData, nil
 }
+
+// splitCertAndKey takes the given data and splits it into cert and key
+// this function doesn't check if the returned cert and key is not empty as this
+// can't be enforced. It is possible the secret in the key vault only contains the
+// cert or key.
+func splitCertAndKey(certAndKey string) (string, string) {
```
Comment (nilekhc) [inline pkg/provider/provider.go:886]: Could we add named returned params for better code readability?

File: .pipelines/templates/e2e-test.yaml (hunk 2 above)
Comment (aramase) [reply .pipelines/templates/e2e-test.yaml:76]: If the image is a released versions (i.e <= `v1.3`), it still doesn't support the split cert/key feature, so we need to skip tests for those versions.

Comment (aramase) [reply .pipelines/templates/e2e-test.yaml:78]: added this as a comment in the file

File: pkg/provider/provider.go (hunk 3 above)
Comment (aramase) [reply pkg/provider/provider.go]: done!

**Review by nilekhc — APPROVED** [review]

**Pull request merged**